app.js

```diff
@@ -113,15 +142,21 @@ app.listen(port, (err) => {
     console.log("App Running");
 });
 
-//Check if the short code is present in the DB
-function checkShortCodeIsUnique(shortCode) {
-    if (db.urls.some(url => url.short === shortCode)){
-        return false;
-    } else {
-        return true;
+async function generateUniqueShortCode() {
+    let short_code = codeGenerator.generateShortUrlCode();
+    let unique = false;
+    while (!unique) {
```

Comment:
Consider using ```await``` and a helper that doesn't throw:
```javascript
while (true) {
    const short_code = codeGenerator.generateShortUrlCode();
    if (await dbUtil.shortCodeAvailable(short_code)) {
        return short_code;
    }
}
```
And ```shortCodeAvailable``` would be ```Promise<boolean```, and could call ```getShortCode``` underneath but just interpret that result to return true/false. (this all assumes you're gonna keep the check & write steps separated)